### PR TITLE
fix(Backend): use object over loose array for permissions

### LINKED ISSUE #1131: Undefined array key "manageTable" at ViewService.php#419
### Steps to reproduce

- `GET /apps/tables/view`

### Expected behavior

Nothing in the logs

### Actual behavior

Undefined array key "manageTable" at …/apps/tables/lib/Service/ViewService.php#419

### Tables app version

0.7.2

### PR DESCRIPTION
**Pull request by blizzz**
Solves issues with undefined array keys, but is said to be more performant but also more clear when coding.

fixes #1131

### CODE REVIEW

File: lib/Db/Table.php [hunk 1]
```diff
@@ -101,10 +102,9 @@ public function jsonSerialize(): array {
 	}
 
 	/**
-	 * @psalm-suppress MismatchingDocblockReturnType
-	 * @return array{read: bool, create: bool, update: bool, delete: bool, manage: bool}|null
+	 * @return ?Permissions
```
Comment (juliushaertl) [inline lib/Db/Table.php:105]: No need as we have the type hint already.

**Review by juliushaertl — APPROVED** [review]

File: lib/Db/Table.php (hunk 1 above)
Comment (blizzz) [reply lib/Db/Table.php]: and removed 2, 3 others of the same

**Comment by blizzz** [comment]
ok, backport does not work without pulling down more things from main.

**Pull request merged**